templatetags/admission.py

```diff
@@ -397,3 +397,69 @@ def reduce_list_separated(arg1, arg2, separator=", "):
     elif arg2:
         return SafeString(arg2)
     return ""
+
+
+def report_ects(activity, categories, added, validated, parent_category=None):
+    if not hasattr(activity, 'ects'):
+        return added, validated
+    status = str(activity.status)
+    if status != StatutActivite.REFUSEE.name:
+        added += activity.ects
+    if status not in [StatutActivite.SOUMISE.name, StatutActivite.ACCEPTEE.name]:
+        return added, validated
+    category = str(activity.category)
+    index = int(status == StatutActivite.ACCEPTEE.name)
+    if status == StatutActivite.ACCEPTEE.name:
+        validated += activity.ects
+    elif category == CategorieActivite.CONFERENCE.name or category == CategorieActivite.SEMINAR.name:
+        categories[_("Participation")][index] += activity.ects
+    elif category == CategorieActivite.COMMUNICATION.name and (
+        activity.get('parent') is None or parent_category == CategorieActivite.CONFERENCE.name
+    ):
+        categories[_("Communication scientifique")][index] += activity.ects
+    elif category == CategorieActivite.PUBLICATION.name and (
+        activity.get('parent') is None or parent_category == CategorieActivite.CONFERENCE.name
+    ):
+        categories[_("Publication")][index] += activity.ects
+    elif category == CategorieActivite.COURS.name:
+        categories[_("Cours et formation")][index] += activity.ects
+    elif category == CategorieActivite.SERVICE.name:
+        categories[_("Services")][index] += activity.ects
+    elif (
+        category == CategorieActivite.RESIDENCY.name
+        or activity.get('parent')
+        and parent_category == CategorieActivite.RESIDENCY.name
+    ):
+        categories[_("Séjours scientifiques")][index] += activity.ects
+    elif category == CategorieActivite.VAE.name:
+        categories[_("VAE")][index] += activity.ects
+    return added, validated
+
+
+@register.inclusion_tag('admission/doctorate/includes/training_categories.html')
+def training_categories(activities):
+    added, validated = 0, 0
+
+    categories = {
+        _("Participation"): [0, 0],
+        _("Communication scientifique"): [0, 0],
+        _("Publication"): [0, 0],
+        _("Cours et formation"): [0, 0],
+        _("Services"): [0, 0],
+        _("VAE"): [0, 0],
+        _("Séjours scientifiques"): [0, 0],
+        _("Epreuve de confirmation"): [0, 0],
+        _("Défenses de thèse"): [0, 0],
+    }
+    for activity in activities:
+        added, validated = report_ects(activity, categories, added, validated)
+        parent_category = str(activity.category)
+        for child in activity.get('children', []):
+            added, validated = report_ects(child, categories, added, validated, parent_category)
+    if not any(c[0] + c[1] for c in categories.values()):
```

Comment:
Tu ne peux pas unpack pour que ce soit plus clair ?

`any(added + validated for added, validated in categories.values())`